windows/hid.c

```diff
@@ -463,28 +463,22 @@ struct hid_device_info HID_API_EXPORT * HID_API_CALL hid_enumerate(unsigned shor
 				cur_dev->path = NULL;
 
 			/* Serial Number */
-			wstr[0]= 0x0000;
+			wstr[0]= L'\0';
 			res = HidD_GetSerialNumberString(write_handle, wstr, sizeof(wstr));
-			wstr[WSTR_LEN-1] = 0x0000;
-			if (res) {
-				cur_dev->serial_number = _wcsdup(wstr);
-			}
+			wstr[WSTR_LEN-1] = L'\0';
+			cur_dev->serial_number = _wcsdup(wstr);
```

Comment:
I don't think I follow the reasoning for this change.
So now, if `HidD_GetSerialNumberString` we would get a NULL-terminated but still garbage, aren't we?

Reply:
`HidD_GetSerialNumberString` doesn't touch buffer if there was error.
So we'll receive null terminated empty string.